packages/components/inputs/async-select-input/src/async-select-input.js

```diff
@@ -35,7 +35,10 @@ const AsyncSelectInput = (props) => {
 
   const placeholder =
     props.placeholder || intl.formatMessage(messages.placeholder);
-
+  const loadingMessage = () =>
+    props.loadingMessage
+      ? props.loadingMessage()
```

Comment:
not enough to have this as string? 
what benefits does it have being a function over a string?

Reply:
The `react-select` accepts a callback. Thats what is it a function

Reply:
We can ask user to send a string and then we change it to a callback when passing it to the react-select's component

Reply:
sounds reasonable to me

Reply:
So I check the `noOptionsMessage` prop, which has similar behaviour as `loadingMessage`. The AsyncSelectInput expects a `func` and the users in the MC always sends a callback. I think we should keep it consistent otherwise it won't look good. But I will change this to expect a string  and hopefully in the future we move the `noOptionsMessage` to string as well

Reply:
consistent in terms of having everything a function?

Reply:
consistent in terms of having similar to what react-select is expecting. i.e. here either both callback functions or both strings

Reply:
Ah, in that case I think it makes sense to keep it the way out `react-select` and our other `Async` selects work.

Doesn't make sense to change the others in this PR and also it would be breaking.

Reply:
of-course it won't be changed. It is big breaking change.

Reply:
it doesn't make sense to change the others, I agree.
if we want to keep consistent towards `react-select` API, sure.

but preferably, it would been great from a consumer side to pass just a string and on our side we forward it as a callback just as you originally suggested.
just my two cents.

Reply:
We could also, for this one, check the type of the passed prop and treat it accordingly. This way, users can keep the same expectation as they have for the other Selects, but are also free to use the "easier" way of just passing the string :)

Reply:
Added support for both `string` and `func` types of `loadingMessage`. This way we can give flexibility of what to use and at the same time be consistent with `react-select`